### ISSUE TITLE
Cannot Use Reflux Components With React getDerivedStateFromProps() Static Method

### ISSUE THREAD

**Issue by tgv1975** [reflux/refluxjs#549]
React 16.4.1
Reflux 6.4.1

---

On components that extend Reflux.Component, if I use

````
static getDerivedStateFromProps(props, state) {...}
````
**without** implementing componentWillMount() (which is now UNSAFE in React), the Reflux functionality in the component fails, and I get a console error:

![image](https://user-images.githubusercontent.com/15803440/42690845-87d2da1a-86ae-11e8-833f-30a3f0c793d4.png)

In the screenshot, below the error, I printed *this.componentWillMount* from the component's render() method, attesting it comes from the Reflux.Component implementation.

In summary, due to implementing a now UNSAFE componentWillMount(), Reflux.Component cannot be used with the new static method getDerivedStateFromProps().

**Comment by TwR-Slice** [comment]
also, my observation is that if you use getDerivedStateFromProps the Store will not mix its state into a Reflux Component.

**Comment by tgv1975** [comment]
@TwR-Slice This library hasn't been updated in more than a year. I have stopped using it since. Probably worth investigating other Redux solutions.

**Comment by TwR-Slice** [comment]
@tgv1975 Sad to see this lib decay, always liked the simplicity of it.
